blocklylib-core/src/main/java/com/google/blockly/android/WorkspaceFragment.java

```diff
@@ -104,4 +104,8 @@ public Workspace getWorkspace() {
     public WorkspaceView getWorkspaceView() {
         return mWorkspaceView;
     }
+
+    public boolean hasBlocks() {
+        return getWorkspace().getRootBlocks().size() > 0;
```

Comment:
This is the logic that used to be called by onRunCode (above). I've extracted it here and it's now called in the BlocklyGenerator (below).